scripts/open-coredump.sh

```diff
@@ -0,0 +1,130 @@
+# The script is reentrant: running it after the prepartory steps will re-use what
+# is already donwloaded. It is *strongly* recommended to run this from an empty
+# directory, with nothing but the core-file present.
+#
+# Requirements:
+# * eu-unstrip
+# * curl
+# * jq
+# * git
+
+set -e
+
+SCYLLA_S3_RELOC_SERVER_URL="http://backtrace.scylladb.com"
+
+if [[ $# != 1 ]]; then
+	echo Please provide the path to the coredump
+	exit 1
+fi
+
+for required in eu-unstrip jq curl git; do
+	if ! type $required >& /dev/null; then
+		echo Please install $required first
+		exit 1
+	fi
+done
+
+COREFILE=$1
+
+BUILD_ID=$(eu-unstrip -n --core ${COREFILE} | cut -f2 -d' ' | cut -f1 -d@)
+
+echo "Build id: ${BUILD_ID}"
+
+BUILD=$(curl -s -X GET "${SCYLLA_S3_RELOC_SERVER_URL}/build.json?build_id=${BUILD_ID}")
+
+RESPONSE_BUILD_ID=$(jq -r .build_id <<< $BUILD)
+VERSION=$(jq -r .version <<< $BUILD)
+PRODUCT=$(jq -r .product <<< $BUILD)
+RELEASE=$(jq -r .release <<< $BUILD)
+ARCH=$(jq -r .arch <<< $BUILD)
+BUILD_MODE=$(jq -r .build_mode <<< $BUILD)
+PACKAGE_URL=$(jq -r .package_url <<< $BUILD)
+
+if [[ "$RESPONSE_BUILD_ID" != "$BUILD_ID" ]]
+then
+    echo "Mismatching build id: requested ${BUILD_ID} but got ${RESPONSE_BUILD_ID}";
+    exit 1
+fi
+
+echo "Matching build is ${PRODUCT}-${VERSION} ${RELEASE} ${BUILD_MODE}-${ARCH}"
+
+PACKAGE_FILE=$(basename ${PACKAGE_URL})
+
+if ! [[ -d ${PRODUCT} ]]
+then
+    if ! [[ -f ${PACKAGE_FILE} ]]
+    then
+        echo "Downloading relocatable package from ${PACKAGE_URL}"
+        curl --output ./${PACKAGE_FILE} ${PACKAGE_URL}
+    else
+        echo "Relocatable package ${PACKAGE_URL} already downloaded"
+    fi
+
+    echo "Extracting package ${PACKAGE_FILE}"
+    tar -zxf $PACKAGE_FILE
+else
+    echo "Relocatable package ${PACKAGE_URL} already downloaded and extracted"
+fi
+
+COMMIT_HASH=$(cut -f3 -d. <<< $RELEASE)
+BASE_VERSION=$(grep -o "^[0-9]\+\.[0-9]\+" <<< $VERSION)
+BRANCH=branch-${BASE_VERSION}
+
+if ! [[ -d ${PRODUCT}.git ]]
+then
+    echo "Cloning ${PRODUCT}.git"
+    git clone -b ${BRANCH} https://github.com/scylladb/${PRODUCT}.git ${PRODUCT}.git
+    cd ${PRODUCT}.git
+    git checkout -q ${COMMIT_HASH}
+    # Skip the other submodules, they are not needed for debugging
+    git submodule update -q --depth=1 --init seastar
+    cd ..
+else
+    echo "${PRODUCT}.git already cloned"
+fi
+
+if ! [[ -f scylla-gdb.py ]]
+then
+    MAIN_BRANCH=master
+    if [[ "${PRODUCT}" == "scylla-enterprise" ]]
+    then
+        MAIN_BRANCH=enterprise
+    fi
+
+    echo "Downloading scylla-gdb.py"
+    curl --output scylla-gdb.py https://raw.githubusercontent.com/scylladb/${PRODUCT}/${MAIN_BRANCH}/scylla-gdb.py
+else
+    echo "scylla-gdb.py already present, not downloading"
+fi
+
+cat << EOF
+Launching dbuild container.
+
+To examine the coredump with gdb:
+
+    $ gdb -q -ex 'set directories scylla.git' --core ${COREFILE} /opt/scylladb/libexec/scylla
```

Comment:
I usually use my host gdb, it's better than the older gdb in dbuild. I have to use solib-search-path and libthread-db-search-path, but it works reliably.

Reply:
My goal with this script was to get a simple case working that is good enough. We all have our preferences and it will be impossible to reconcile all of them in a script.
The best I can do is to provide opportunity for the user keep using their preferred way of debugging while still using the script. Which is one of the reasons I don't automatically start gdb with the script.
Some will want to plug in their `.gdbinit`, others want to use host `gdb`, still others might prefer this or that flag. The script shouldn't assume, it should give opportunity.

Reply:
I will emphasize this in the cover letter, because I got many such suggestions so far.